common/lcs/src/ser.rs

```diff
@@ -197,7 +192,7 @@ impl<'a> ser::Serializer for &'a mut Serializer {
     where
         T: ?Sized + Serialize,
     {
-        self.serialize_u8(1)?;
+        self.output.push(1);
```

Comment:
This seems like a step back in terms of readability, we should probably try to ensure that all writing of data goes through the "serialize" methods.

Reply:
`serialize_u8` is a user-facing API that consumes `self`. If needed, we should be able to create internal helper function, though.